core/src/main/java/org/elasticsearch/search/aggregations/bucket/children/ParentToChildrenAggregator.java

```diff
@@ -225,8 +225,9 @@ public ChildrenAggregatorBuilder(String name, String childType) {
             if (childDocMapper != null) {
                 ParentFieldMapper parentFieldMapper = childDocMapper.parentFieldMapper();
                 if (!parentFieldMapper.active()) {
+                    // NOCOMMIT fix exception args
                     throw new SearchParseException(aggregationContext.searchContext(),
```

Comment:
I think it should rather be an illegalargumentexception since the problem is with the request (so that it would be a 4xx code instead of a 5xx code).